plugins/datetime/msd-datetime-mechanism.c

```diff
@@ -113,331 +160,334 @@ msd_datetime_mechanism_error_get_type (void)
-{
-        MsdDatetimeMechanism      *mechanism;
-
-        mechanism = MSD_DATETIME_MECHANISM (G_OBJECT_CLASS (msd_datetime_mechanism_parent_class)->constructor (
-                                                type,
-                                                n_construct_properties,
-                                                construct_properties));
-
-        return G_OBJECT (mechanism);
-}
-
-static void
-msd_datetime_mechanism_class_init (MsdDatetimeMechanismClass *klass)
+static gboolean
+msd_datetime_mechanism_adjust_time_handler (MateSettingsDateTimeMechanism *object,
+                                            GDBusMethodInvocation         *invocation,
+                                            gint64                         seconds_to_add,
+                                            gpointer                       user_data)
 {
-        GObjectClass   *object_class = G_OBJECT_CLASS (klass);
+        struct timeval tv;
+        gboolean ret = FALSE;
+        GError *error = NULL;
+        MsdDatetimeMechanism *mechanism;
 
-        object_class->constructor = msd_datetime_mechanism_constructor;
-        object_class->finalize = msd_datetime_mechanism_finalize;
+        mechanism = MSD_DATETIME_MECHANISM (user_data);
 
-        dbus_g_object_type_install_info (MSD_DATETIME_TYPE_MECHANISM, &dbus_glib_msd_datetime_mechanism_object_info);
+        reset_killtimer ();
+        g_debug ("AdjustTime(%ld) called", seconds_to_add);
 
-        dbus_g_error_domain_register (MSD_DATETIME_MECHANISM_ERROR, NULL, MSD_DATETIME_MECHANISM_TYPE_ERROR);
+        if (gettimeofday (&tv, NULL) != 0) {
+                error = g_error_new (MSD_DATETIME_MECHANISM_ERROR,
+                                     MSD_DATETIME_MECHANISM_ERROR_GENERAL,
+                                     "Error calling gettimeofday(): %s", strerror (errno));
+                g_dbus_method_invocation_return_gerror (invocation, error);
+                g_error_free (error);
+                return FALSE;
+        }
 
-}
+        if (!_check_polkit_for_action (mechanism,
+                                       "org.mate.settingsdaemon.datetimemechanism.settime",
+                                       g_dbus_method_invocation_get_sender (invocation),
+                                       &error)) {
+                g_dbus_method_invocation_return_gerror (invocation, error);
+                g_error_free (error);
+                return FALSE;
+        }
 
-static void
-msd_datetime_mechanism_init (MsdDatetimeMechanism *mechanism)
-{
-        mechanism->priv = msd_datetime_mechanism_get_instance_private (mechanism);
+        tv.tv_sec += (time_t) seconds_to_add;
+        ret = _set_time (mechanism, &tv, &error);
+        if (!ret) {
+                g_dbus_method_invocation_return_gerror (invocation, error);
+                g_error_free (error);
+        } else {
+                mate_settings_date_time_mechanism_complete_adjust_time ( object, invocation);
```

Comment:
one unnecessary space after bracket.
```suggestion
                mate_settings_date_time_mechanism_complete_adjust_time (object, invocation);
```

Reply:
Done.